packages/nuxt/src/core/runtime/nitro/renderer.ts

```diff
@@ -157,7 +157,12 @@ const getSPARenderer = lazyCachedFunction(async () => {
 
 async function getIslandContext (event: H3Event): Promise<NuxtIslandContext> {
   // TODO: Strict validation for url
-  const url = event.node.req.url?.substring('/__nuxt_island'.length + 1) || ''
+  let url = event.node.req.url || ''
+  if (process.env.prerender && event.node.req.url && await useStorage('internal:nitro:prerender:island-props').hasItem(event.node.req.url)) {
+    // rehydrate props from cache so we can rerender island if cache does not have it any more
+    url = await useStorage('internal:nitro:prerender:island-props').getItem(event.node.req.url) as string
```

Comment:
Maybe use prefixed storages to keep code readable?

Reply:
ahh ... it does actually increase the size.